Correcting an accident merge to main

BenWehrend opened the pull request:
Closes #40

I accidentally merged main on my end, so I re-checked out to my branch the changes I made are:

Adding these two querySelectors:

domUpdates.js line 47-49
// clear buttons query selectors
const clearSearchBox = document.querySelector(".clear-search");
const clearTags = document.querySelector(".clear-tags");


That match to these functions to these buttons:

domUpdates.js line 148-160
clearSearchBox.addEventListener('click', function() {
  searchBox.value = '';
  filterRecipes();
});

clearTags.addEventListener('click', function() {
  const activeTags = tagsContainer.querySelectorAll('.tag-active');
  activeTags.forEach((tag) => {
      tag.classList.remove('tag-active');
  });

  filterRecipes();
});
Along with the respective HTML and CSS changes:

index.html lines 26-27
        <button class="clear-search">Clear Search</button>
        <button class="clear-tags">Clear Tags</button>


style.scss lines 107-122        
  .random-recipe,
  .clear-search,
  .clear-tags {
    background-color: #d39738;
    color: white;
    font-size: 1.25rem;
    padding: 10px;
  }

  .tag,
  .random-recipe,
  .clear-search,
  .clear-tags {
    font-family: sans-serif;
    border: none;
    border-radius: 4px;
  }

BenWehrend committed:
adds button for clearing search & tags, styling w.i.p.

BenWehrend committed:
fixes a mistake

KojinKuro commented:
Good job linking the issue to the PR! I think the functionality is all there so you won't need to change any of that. I have two bits of feedback.

### The Design (focus on fixing this)
I think the buttons are too big and mess with the visual hierarchy of the page. The clear search button takes up more weight on the screen than the search bar itself. I think it would be better if you made them less of a focus in the filtering section. I mocked up something quick that you can follow.

![image](https://github.com/KojinKuro/whats-cookin/assets/11234292/67b94e33-6b94-4108-8841-abf05d9573ac)

### Event Delegation (less important)
This second bit of feedback is a nit but I was wondering if you could merge `clearSearchBox.addEventListener()` and `clearTags.addEventListener()` together with the following code using event delegation. That way we have less event listeners and keep our website using less memory. This might make the file less organized so I'll let you decide that. 

At the very least I think it would be good merging `clearSearchBox.addEventListener()` and `clearTags.addEventListener()` into one eventListener so that way: a.) it uses less memory b.) they are grouped together for clarity.

https://github.com/KojinKuro/whats-cookin/blob/2d989ac865967560042442ba03e17d9a47111d29/src/domUpdates.js#L147-L158

https://github.com/KojinKuro/whats-cookin/blob/2d989ac865967560042442ba03e17d9a47111d29/src/domUpdates.js#L58-L63

BenWehrend committed:
adjusts layout and JS to use event delegation

BenWehrend committed:
fixes html layout issue

BenWehrend committed:
finishes off clear tags visibility

BenWehrend committed:
merges repetitive clearseacrh eventListener into filterSettings

BenWehrend commented:
I made the adjustments, hopefully this is what you're looking for. Main thing I changed was changing the tagsContainer eventListener to the filterSettings eventListener to encapsulate functionality with event delegation. Also added toggleClearButtons function

KojinKuro committed:
fix: style issues and visual feedback

KojinKuro committed:
refactor: Remove repetive code and keept HTML CSS free

KojinKuro commented:
Good job fitting everything in one event listener and matching the design! I added a hover effect on `clear tags` for visual feedback and changing the font size of the search bar from using screen size to rem because it felt weird that the search button was scaling with my screen size. 

I also refactored your code to be more DRY and SRP. I also changed from changing the style of the element to be hidden or visible to toggle a class. This is to keep the HTML style free and make it more clear what things are doing. 

Looking through out the code I realized that there are other parts that could be more dry but that is a separate issue we can open in the future. Merging into main now.
